### ISSUE TITLE
iOS 12.1.4 Apple A8 iPhone 6+ "Unable to find kernel's credentials in kernel memory"

### ISSUE THREAD

**Issue by matwlodek** [pwn20wndstuff/Undecimus#1204]
**Describe the bug**
Jailbreak fails at three states in my device:
1. [*] got gc at 1 -- breaking -  (1/31) my iPhone is rebooting
2. [*] KernelUtilities.c(294): Assertion failure - sometimes my iPhone reboots at the second Assertion failure
3. [*] Unable to find kernel's credentials in kernel memory - (2/31) several tries and reboots and I did not go pass this 

**To Reproduce**
1. Turn off, then on iPhone.
2. Run unc0ver.
3. Click "Jailbreak"
4. See error after several tries.

**Screenshots**
![IMG_0003](https://user-images.githubusercontent.com/24829691/61575443-2b6b1880-aacc-11e9-8487-4e1f7d181855.PNG)
![IMG_0004](https://user-images.githubusercontent.com/24829691/61575444-2dcd7280-aacc-11e9-84e7-e437b85dc330.PNG)
![IMG_0005](https://user-images.githubusercontent.com/24829691/61575445-30c86300-aacc-11e9-8eae-526b3e298612.PNG)

**Device (please complete the following information):**
 - iOS Version: iOS 12.1.4 (Stable) (Build: 16D57)
 - iOS Device: iPhone 6+
 - unc0ver Version: 3.3.7

**Place an "x" between the brackets if true:**
 - [x] this is a bug others will be able to reproduce
 - [ ] this issue is present with all tweaks uninstalled(except for default packages) or disabled
 - [x] this issue is present after a rootfs restore
 - [x] this issue is present on the latest version of unc0ver

**Logs**
[*] unc0ver Version: 3.3.7
[*] Machine Name: iPhone7,1
[*] Model Name: N56AP
[*] Kernel Version: Darwin Kernel Version 18.2.0: Wed Dec 19 20:28:48 PST 2018; root:xnu-4903.242.2~ 1/RELEASE_ARM64_T7000
[*] Kernel Page Size: 0x1000
[*] System Version: iOS 12.1.4 (Stable) (Build: 16D57)
[*] Bundled Resources Version: 1.0~b23
[*] Status: Jailbreak
[*] Progress: Ready to jailbreak
[*] Status: Jailbreaking
[*] KernelUtilities.c(1461): Assertion failure
[*] Progress: Exploiting kernel...
[*] kCFCoreFoundationVersionNumber: 1562.000000
[*] offsets selected for iOS 12.0 or above
[*] got gc at 1 -- breaking
[*] tfp0: 0x1707
[*] kernel_base: 0xfffffff0152f1000
[*] kernel_slide: 0x000000000e2ed000
[*] Successfully exploited kernel.
[*] Progress: Initializing jailbreak...
[*] Finding kernel offsets...
[*] Detected monolithic kernel.
[*] Escaping sandbox...
[*] myProcAddr = 0xfffffff0575c7be8
[*] KernelUtilities.c(294): Assertion failure
[*] kernelCredAddr = 0x0000000000000000
[*] Unable to find kernel's credentials in kernel memory.

**Comment by husui007** [comment]
I am using an iPhone 6 and have the same issue.  

KernelUtilities.c(1461): Assertion failure
kCFCoreFoundationVersionNumber: 1562,000000
offsets selected for iOS 12.0 or above
got gc at 2 -- breaking

After about 3-4 seconds my iPhone will reboot.  

I tried to jailbreak using Cydia Impactor using unc0ver 3.3.7.  Then tried with Tweakbox using unv0ver 3.3.7.  

Find my phone was off
Airplane mode was on
PIN code was off

My settings are similar to except Reload Daemons was set to off.  

Not sure what else to try to get this working.  Any hints?

**Comment by djares08** [comment]
I am using an iPad mini 3 cellular and have this issue.

KernelUtilities.c(1494): Assertion failure
kCFCoreFoundationVersionNumber: 1561,000000
offsets selected for iOS 12.0 or above
Tfp0:0x3517
Kernel_base 0xfffffff011604000
Kernel_slide:0x000000000a600000
Successfully exploited kernel.
Progress: Initializing jailbreak...
Finding kernel offsets....

After this step my iPad reboot.

I’m using unc0ver 3.3.8.

**Comment by 0x9fff00** [comment]
I'm getting this on iPhone 8 (iPhone10,4) on 12.2 using unc0ver 3.4.2.

**Comment by anorgeot** [comment]
Same issue. iPhone 6. Unc0ver 3.4.2.

**Comment by matwlodek** [comment]
3.4.0 & 3.4.2 fixed everything for me. But back then on 3.3.7 I have performed factory reset from phone and set it up as new iPhone. So I think that the factory reset and newer version of unc0ver is a fix for this issue. Try it out, maybe it will work for you too.

**Comment by 0x9fff00** [comment]
Update: Just tried 3.3.8 (Sock Port) which worked, and now also 3.4.2 works, even after jumping to orig-fs in SnapBack. So at least in my case it seems to only have affected the initial jailbreak.

**Comment by ghost** [comment]
I restored my device (iPhone X) using future restore from 12.1.1 to 12.2 and then restored from backup.
Faced the same issue when trying to jailbreak for the first time after restoring with 3.4.2.
Tried 3.3.8 Sock Port and everything went smoothly without any errors. Issue seem to be affecting only first time jailbreakers or users who restored their devices.
P.S My device was previously jailbroken on 12.1.1 and I restored FS before backing up.

**Comment by markcoker** [comment]
Have you tried reinstalling unc0ver on iPhone?

**Comment by anorgeot** [comment]
Successful jailbreak after install 3.5.2 for me. Thanks

**Comment by rolltidehero** [comment]
I had the same issue on 3.5.5 on a fresh iOS 12.4 install.

**Workaround: The 3.5.2 release worked fine for me.**

**Comment by ipatch** [comment]
iphone 6 here.

Upgraded to 12.4 from 12.2, was jailbroken using unc0ver with 12.2 seldom had issues.

I ran into this issue after upgrading to 12.4 from a fresh download of 12.4 from ipsw.me and installing 12.4 from itues, then tried jailbreaking 12.4 after setting up 12.4 from a itunes backup after 12.4 installed via itunes, ran into several issues with unc0ver however i don't think unc0ver is the issue here but more than likely this is a rootfs issue because I performed the itunes backup while my phone was jailbroken so maybe there was / is rootfs alignment issues when trying to jb 12.4 that said. if running into this issue it's working trying to use the **restore rootfs** toggle in unc0ver then run the jailbreak again. Used 3.5.6 and took a reboot or two to get going and took ~ 60 seconds to jailbreak for whatever reason whereas it was taking about ~ 10 seconds on 12.2

Hope future readers _myself included_ find this useful

cheers 🍩
